src/helpers/parser.ts

```diff
@@ -25,19 +26,32 @@ export const gitIssueParser = async ({ owner, repo, issue_number }: GitParser):
   }
 };
 
-export const gitLinkedIssueParser = async ({ owner, repo, issue_number }: GitParser): Promise<string> => {
+export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitParser): Promise<string> => {
   try {
-    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${issue_number}`);
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${pull_number}`);
     const dom = parse(data);
     const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
-    const linkedPRs = devForm.querySelectorAll(".my-1");
+    const linkedIssues = devForm.querySelectorAll(".my-1");
 
-    if (linkedPRs.length === 0) {
+    if (linkedIssues.length === 0) {
       return "";
     }
 
-    const prUrl = linkedPRs[0].querySelector("a")?.attrs?.href || "";
-    return prUrl;
+    const issueUrl = linkedIssues[0].querySelector("a")?.attrs?.href || "";
+    return issueUrl;
+  } catch (error) {
+    return "";
+  }
+};
+
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser): Promise<string> => {
+  try {
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/issues/${issue_number}`);
+    const dom = parse(data);
+    const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
+    const linkedPRs = devForm.querySelectorAll(".my-1");
+    if (linkedPRs.length === 0) return "";
+    return linkedPRs[linkedPRs.length - 1].querySelector("a")?.attrs?.href || "";
```

Comment:
Take a look at issue https://github.com/ubiquity/ubiquibot/issues/583. You can see 4 linked PRs, 2 of them for ubiquibot repo and 2 of them is in assignee's personal repo. The correct PR that was merged is in 2nd place, so we should not rely on ordering.

Please make sure the function only finds a merged PR that was merged in the correct repository.

Reply:
Okay, So 'll check the pr if its org name and repo name is same as the issue's org & repo.

And also, if there are multiple merged prs, we will only get the most recent one.

@pavlovcik , @whilefoo Rfc

Reply:
Keep it simple: 

When the rewards are generated, look for the most recent merged pull request (check the time stamp)

Just use that one.

Reply:
No you should not go based on the pull request number. You should check when the pull request had a state change from "open" to "merged"

Reply:
Ooo right. Got it.